packages/wpk/generate_wpk_package.sh

```diff
@@ -0,0 +1,381 @@
+                    shift 2
+                elif [[ "${2}" == "aarch64" ]]; then
+                    local ARCHITECTURE="${2}"
+                    local LINUX_BUILDER="${LINUX_BUILDER_AARCH64}"
+                    local LINUX_BUILDER_DOCKERFILE="${LINUX_BUILDER_AARCH64_DOCKERFILE}"
+                    shift 2
+                    echo "Architecture not supported currently. Issue: https://github.com/wazuh/wazuh/issues/22400"
+                    help 1
+                elif [[ "${2}" == "armv7hl" ]]; then
+                    local ARCHITECTURE="${2}"
+                    local LINUX_BUILDER="${LINUX_BUILDER_ARMV7HL}"
+                    local LINUX_BUILDER_DOCKERFILE="${LINUX_BUILDER_ARMV7HL_DOCKERFILE}"
+                    shift 2
+                    echo "Architecture not supported currently. Issue: https://github.com/wazuh/wazuh/issues/22400"
+                    help 1
+                else
+                    echo "Architecture must be x86_64/amd64, aarch64 or armv7hl"
+                    help 1
+                fi
+            else
+              echo "ERROR: Missing architecture."
+              help 1
+            fi
+            ;;
+        "-j"|"--jobs")
+            if [ -n "${2}" ]; then
+                local JOBS="${2}"
+                shift 2
+            else
+                echo "ERROR: Missing jobs."
+                help 1
+            fi
+            ;;
+        "-p"|"--path")
+              if [ -n "${2}" ]; then
+                  INSTALLATION_PATH="${2}"
+                  shift 2
+              else
+                  help 1
+              fi
+              ;;
+        "-pn"|"--package-name")
+            if [ -n "${2}" ]; then
+                local HAVE_PKG_NAME=true
+                local PKG_NAME="${2}"
+                PKG_PATH=`echo ${PKG_NAME}| rev|cut -d'/' -f2-|rev`
+                PKG_NAME=`basename ${PKG_NAME}`
+                shift 2
+            else
+                echo "ERROR: Missing package name"
+                help 1
+            fi
+            ;;
+        "-o"|"--output")
+            if [ -n "${2}" ]; then
+                local HAVE_OUT_NAME=true
+                local OUT_NAME="${2}"
+                shift 2
+            else
+                echo "ERROR: Missing output name."
+                help 1
+            fi
+            ;;
+        "--aws-wpk-key")
+            if [ -n "${2}" ]; then
+                local HAVE_WPK_KEY=true
+                local WPK_KEY="${2}"
+                shift 2
+            fi
+            ;;
+        "--aws-wpk-cert")
+            if [ -n "${2}" ]; then
+                local HAVE_WPK_CERT=true
+                local WPK_CERT="${2}"
+                shift 2
+            fi
+            ;;
+        "--aws-wpk-key-region")
+            if [ -n "${2}" ]; then
+                local AWS_REGION="${2}"
+                shift 2
+            fi
+            ;;
+        "-c"|"--checksum")
```

Comment:
The current logic assumes that if there's an argument after -c or --checksum, it must be the checksum directory. However, if another option follows with its own argument (like -s /tmp), the script will attempt to use /tmp as the checksum directory, leading to an error.

Reply:
added new control to this parameters, but we want to maintain checksum dir parameter i think